### PR TITLE
feat(ui): toasts rework

### LINKED ISSUE #4637: [bug]: Queue with a Shared Setting that causes Errors produce large number of error messages
### Is there an existing issue for this?

- [X] I have searched the existing issues

### OS

Windows

### GPU

cuda

### VRAM

24

### What version did you experience this issue on?

3.2rc2

### What happened?

If a large number of queue items are added which share a model that is found to be invalid (e.g., model config exists, but the model is gone or malformed, etc.), then the front-end is hit with a rapid succession of error messages.

### Screenshots

_No response_

### Additional context

_No response_

### Contact Details

_No response_

### LINKED ISSUE #5136: [enhancement]: prevent duplicate toasts
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Contact Details

_No response_

### What should this feature add?

We have some situations where a lot of duplicate toasts can pop up. The most egregious scenario is when you queue up a lot of gens and they all have a fatal error. You get a bajillion toasts.

This will require some broad changes.

Probably we should do this anyways, as currently the toasts are implemented kinda wonkily, with at least 3 ways to create toasts:
- Using Chakra's own hook
- Using a custom hook that wraps chakra's
- By dispatching a toast action to redux, which is handled via extremely naive redux-based toast "queue" (more of a stack). This was created because I didn't realize you can dispatch toasts from outside a component. **It should be removed**.

We also have a `makeToast` function that provides some sensible defaults, returning a toast config object.

Chakra provides some useful utilities to action this request:
- [prevent duplicate toasts](https://chakra-ui.com/docs/components/toast#preventing-duplicate-toast)
- [update existing toasts](https://chakra-ui.com/docs/components/toast#updating-toasts)
- [create toasts outside a component](https://chakra-ui.com/docs/components/toast#standalone-toasts)

We should design a simple API for creating toasts, and implement it as a wrapper for the standalone toast util and separately as a wrapper for the toast hook. We may want to introduce our own simple state management using `nanostores` so we can manage toasts anywhere in the JS, if the chakra utils are not sufficient.

### Alternatives

_No response_

### Additional Content

_No response_

### PR DESCRIPTION
**Pull request by psychedelicious**
## Summary

Add a utility for managing toasts. A single method `toast` is exported. It can be used to both create a new toast or update an existing one, if provided the ID of an existing toast. All chakra toast options are supported, plus one new option: `withCount`.

`withCount` defaults to true, and handles appending a couple to the title of the toast when mutltiples are fired:
```ts
toast({ title: 'Hello', withCount: true });
// first toast title: 'Hello'
// second toast title: 'Hello (2)'
// third toast title: 'Hello (3)'
``` 

When updating a toast, the new title, description and other params will immediately override the existing toast.

If set to false, no count is appended. Note that if the id changes each time the toast is fired, that is considered a different toast and it will create a new one instead of updating.

All toasts across the app have been revised to use this utility. There's no more "toast queue" - toasts are fired the same way whether in a component, listener or wherever.

Other changes:
- Fixed recalling seed not showing a toast
- Consolidated some model install logic that used toasts, was duplicated like several times

## Related Issues / Discussions

- Supersedes #6409 - I copied the component and handling of invocation error events from this PR, thanks @maryhipp 
- Closes #5136
- Closes #4637

## QA Instructions

Note: There is some minor jank with how the toasts resize if you update them rapidly. This is a chakra issue - will need to dig into chakra theming to fix it.

- Do something that makes a toast many times in a row. For example, spam one of the parameter recall buttons.
- Create a graph with a single divide integer node. Leave the values at zero to create a divide-by-zero error. Increase iterations to 5 or so and invoke. You should see one batch queued toast and one error toast that rapidly updates.

## Merge Plan

n/a

## Checklist

- [x] _The PR has a short but descriptive title, suitable for a changelog_
- [ ] _Tests added / updated (if applicable)_ This needs proper e2e tests to test, but I don't think this particular functionality is super important to test.
- [x] _Documentation added / updated (if applicable)_

### CODE REVIEW

File: invokeai/frontend/web/src/features/toast/toast.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+   * @example
+   * toast({ title: 'Hello', withCount: true });
+   * // first toast: 'Hello'
+   * // second toast: 'Hello (2)'
+   */
+  withCount?: boolean;
+};
+
+/* eslint-disable @typescript-eslint/no-explicit-any */
+// Any is correct here; we accept anything as toast data parameter.
```
Comment (maryhipp) [inline invokeai/frontend/web/src/features/toast/toast.ts:28]: Is this comment out of date?

File: invokeai/frontend/web/src/app/store/middleware/listenerMiddleware/listeners/socketio/socketInvocationError.ts [hunk 2]
```diff
@@ -21,6 +25,30 @@ export const addInvocationErrorEventListener = (startAppListening: AppStartListe
...
+
+      if (error_type === 'OutOfMemoryError') {
+        toast({
+          id: 'INVOCATION_ERROR',
+          title: t('toast.outOfMemoryError'),
+          status: 'error',
+          description: ToastWithSessionRefDescription({
+            message: t('toast.outOfMemoryDescription'),
+            sessionId,
+          }),
```
Comment (maryhipp) [inline invokeai/frontend/web/src/app/store/middleware/listenerMiddleware/listeners/socketio/socketInvocationError.ts:39]:
```suggestion
          }),
          duration: null,
```

File: invokeai/frontend/web/src/app/store/middleware/listenerMiddleware/listeners/socketio/socketInvocationError.ts [hunk 3]
```diff
@@ -21,6 +25,30 @@ export const addInvocationErrorEventListener = (startAppListening: AppStartListe
...
+        });
+      } else {
+        toast({
+          id: `INVOCATION_ERROR_${errorType}`,
+          title: t('toast.serverError'),
+          status: 'error',
+          description: ToastWithSessionRefDescription({
+            message: errorType,
+            sessionId,
+          }),
```
Comment (maryhipp) [inline invokeai/frontend/web/src/app/store/middleware/listenerMiddleware/listeners/socketio/socketInvocationError.ts:49]:
```suggestion
          }),
          duration: null,
```

**Comment by maryhipp** [comment]
This looks good but it will be really hard to copy the session ID if you have a long queue and they are all failing because it basically flickers through each one as the socket error fires. I played around a bit to see if I could easily to just show the first, or a list but no luck. What do you think would be the best UX?

File: invokeai/frontend/web/src/features/toast/ToastWithSessionRefDescription.tsx [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+export default function ToastWithSessionRefDescription({ message, sessionId }: Props) {
+  return (
+    <Flex flexDir="column">
+      <Text fontSize="md">{message}</Text>
+      <Flex gap="2" alignItems="center">
+        <Text fontSize="sm">{t('toast.sessionRef', { sessionId })}</Text>
+        <IconButton
+          size="sm"
+          aria-label="Copy"
+          icon={<PiCopyBold />}
```
Comment (maryhipp) [inline invokeai/frontend/web/src/features/toast/ToastWithSessionRefDescription.tsx:20]:
```suggestion
          icon={<PiCopyBold style={{ fill: 'white' }} />}
```

File: invokeai/frontend/web/src/features/toast/toast.ts (hunk 1 above)
Comment (psychedelicious) [reply invokeai/frontend/web/src/features/toast/toast.ts:28]: Yep

**Comment by psychedelicious** [comment]
@maryhipp I've made the suggested changes, and updated the logic for what the toast displays per our discussion

**Review by maryhipp — APPROVED** [review]

**Pull request merged**